deepcell_spots/applications/polaris.py

```diff
@@ -23,227 +23,132 @@
-            image=resized_image, batch_size=batch_size,
-            pad_mode=pad_mode, preprocess_kwargs=preprocess_kwargs
-        )
-
-        # Resize output_images back to original resolution if necessary
-        label_image = self._resize_output(output_images, image.shape)
+    def __init__(self,
+                 segmentation_model=None,
+                 spots_model=None):
+        if segmentation_model is None:
+            model_path = '../notebooks/models/CytoplasmSegmentation'
+            segmentation_model = tf.keras.models.load_model(model_path)
 
-        # Postprocess predictions to create label image
-        predicted_spots = self._postprocess(label_image, **postprocess_kwargs)
+        self.spots_app = SpotDetection(model=spots_model)
+        self.segmentation_app = CytoplasmSegmentation(model=segmentation_model)
 
-        return predicted_spots
+        self.segmentation_app.preprocessing_fn = histogram_normalization
+        self.segmentation_app.postprocessing_fn = deep_watershed
 
     def predict(self,
                 image,
-                batch_size=4,
                 image_mpp=None,
-                pad_mode='reflect',
-                preprocess_kwargs=None,
-                postprocess_kwargs=None,
-                threshold=0.9):
-        """Generates a list of coordinate spot locations of the input
-        running prediction with appropriate pre and post processing
-        functions.
+                # segmentation
+                cytoplasm_channel=0,
+                # spots
+                spots_channel=1,
+                threshold=0.95,
+                clip=False):
+        """Generates prediction output consisting of a labeled cell segmentation image,
+        detected spot locations, and a dictionary of spot locations assigned to labeled
+        cells of the input.
+
         Input images are required to have 4 dimensions
-        ``[batch, x, y, channel]``.
+        ``[batch, x, y, channel]``. Channel dimension should be 2.
+
         Additional empty dimensions can be added using ``np.expand_dims``.
+
         Args:
             image (numpy.array): Input image with shape
                 ``[batch, x, y, channel]``.
-            batch_size (int): Number of images to predict on per batch.
             image_mpp (float): Microns per pixel for ``image``.
-            pad_mode (str): The padding mode, one of "constant" or "reflect".
-            preprocess_kwargs (dict): Keyword arguments to pass to the
-                pre-processing function.
-            postprocess_kwargs (dict): Keyword arguments to pass to the
-                post-processing function.
+            cytoplasm_channel (int): Value should be 0 or 1 depending on the channel
+                containing the images for cell segmentation. Defaults to 0.
+            spots_channel (int): Value should be 0 or 1 depending on the channel
+                containing the images for spot detection. Defaults to 1.
```

Comment:
Consider allowing images with more than 2 channels and instead check that the channel indicies are valid for the image that is submitted.